lib/ex_webrtc/peer_connection.ex

```diff
@@ -1420,8 +1418,16 @@ defmodule ExWebRTC.PeerConnection do
       dtls_role = if dtls_role in [:actpass, :passive], do: :active, else: :passive
       DTLSTransport.start_dtls(state.dtls_transport, dtls_role, peer_fingerprint)
 
+      # ice_creds will be nil if all of the mlines in the description are rejected
+      # in such case, if this is the first remote description, connection won't be established
       # TODO: this can result in ICE restart (when it should, e.g. when this is answer)
-      :ok = state.ice_transport.set_remote_credentials(state.ice_pid, ice_ufrag, ice_pwd)
+      case ice_creds do
+        nil ->
+          :ok
+
+        {ice_ufrag, ice_pwd} ->
+          :ok = state.ice_transport.set_remote_credentials(state.ice_pid, ice_ufrag, ice_pwd)
+      end
 
       for candidate <- SDPUtils.get_ice_candidates(sdp) do
```

Comment:
I think I would move this inside case, when we have ICE credentials, even though that ICE will ignore remote candidates when it does not have them.